api/src/main/java/org/openmrs/module/eptsreports/reporting/reports/SetupIntensiveMonitoringReport.java

```diff
@@ -0,0 +1,101 @@
+package org.openmrs.module.eptsreports.reporting.reports;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.Properties;
+import org.openmrs.Location;
+import org.openmrs.module.eptsreports.reporting.library.cohorts.GenericCohortQueries;
+import org.openmrs.module.eptsreports.reporting.library.datasets.QualityImprovement2020DataSet;
+import org.openmrs.module.eptsreports.reporting.reports.manager.EptsDataExportManager;
+import org.openmrs.module.eptsreports.reporting.utils.EptsReportUtils;
+import org.openmrs.module.reporting.evaluation.parameter.Mapped;
+import org.openmrs.module.reporting.evaluation.parameter.Parameter;
+import org.openmrs.module.reporting.report.ReportDesign;
+import org.openmrs.module.reporting.report.definition.ReportDefinition;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SetupIntensiveMonitoringReport extends EptsDataExportManager {
+
+  @Autowired protected GenericCohortQueries genericCohortQueries;
+
```

Comment:
Instead of having many @Autowired components, can we adopt the standard of creating those object in a public constructor? this will help faster loading because we will only have one @Autwired object at the method signature

Reply:
done